### PR TITLE
Native dialogs

### LINKED ISSUE #174: Replace Java-based open dialog with OS native dialog
### Environment
Environment
MegaMekLab 0.44.0
OS X 10.13.5
Java 8 update 171

### Description
File -> Load -> From File uses a Java-based open dialog; it would be preferable to use OS Native dialogs.

### PR DESCRIPTION
**Pull request by neoancient**
Replaces use of Swing's cross-platform JFileChooser with the AWT FileDialog, which uses the OS-native dialog. This was already being used for saving and for opening fluff images.

This might best be reviewed one commit at a time. The first was a code cleanup. The next (after the upstream merge) involved tokenizing all the string literals. The third replaced the dialog.

I looked at replacing the cross-platform print dialog with a native one but that is a bit more involved and best left as a separate change.

Fixes #174

### CODE REVIEW

File: src/megameklab/com/util/MenuBarCreator.java [hunk 1]
```diff
@@ -607,198 +591,119 @@ private void jMenuGetUnitWeightBreakdownFromCache_actionPerformed() {
...
-        JFileChooser f = new JFileChooser(filePathName);
-        f.setLocation(parentFrame.getLocation().x + 150, parentFrame.getLocation().y + 100);
-        f.setDialogTitle("Choose Unit");
-        f.setDialogType(JFileChooser.OPEN_DIALOG);
-        f.setMultiSelectionEnabled(false);
-
-        FileNameExtensionFilter filter = new FileNameExtensionFilter("Unit Files", "blk", "mtf", "hmp");
+    private @Nullable
+    File loadUnitFile() {
+        String filePathName = System.getProperty("user.dir") + "/data/mechfiles/";
```
Comment (sixlettervariables) [inline src/megameklab/com/util/MenuBarCreator.java]: Does Java have a path combiner? I know in .NET we'd do: `Path.Combine(System.getProperty("user.dir"), "data", "mechfiles")` and never care about path separators.

File: src/megameklab/com/util/MenuBarCreator.java [hunk 2]
```diff
@@ -817,60 +722,58 @@ private void jMenuInsertImageFile_actionPerformed() {
 
             if (fDialog.getFile() != null) {
                 String relativeFilePath = new File(fDialog.getDirectory() + fDialog.getFile()).getAbsolutePath();
-                relativeFilePath = "." + File.separatorChar + relativeFilePath.substring(new File(System.getProperty("user.dir").toString()).getAbsolutePath().length() + 1);
+                relativeFilePath = "." + File.separatorChar + relativeFilePath.substring(new File(System.getProperty("user.dir")).getAbsolutePath().length() + 1);
```
Comment (sixlettervariables) [inline src/megameklab/com/util/MenuBarCreator.java]: Likewise, is there a relative path tool? (granted this code has probably worked just fine for years)

File: src/megameklab/com/util/MenuBarCreator.java [hunk 3]
```diff
@@ -1390,4 +1289,8 @@ public void lostOwnership(Clipboard arg0, Transferable arg1) {
 
     }
 
+    private final FilenameFilter unitFilesFilter =
```
Comment (sixlettervariables) [inline src/megameklab/com/util/MenuBarCreator.java]: Do these endsWith need to be case sensitive?

Comment (neoancient) [reply src/megameklab/com/util/MenuBarCreator.java]: Good catch. I'll need to fix that.

File: src/megameklab/com/util/MenuBarCreator.java (hunk 2 above)
Comment (neoancient) [reply src/megameklab/com/util/MenuBarCreator.java]: All I was doing here was removing the extraneous `toString()` call on a String expression. I didn't worry about the rest of the line, which has proved to be functional if not necessarily ideal.

File: src/megameklab/com/util/MenuBarCreator.java (hunk 1 above)
Comment (neoancient) [reply src/megameklab/com/util/MenuBarCreator.java]: The java.nio.file.* package has tools that do that sort of thing, but I don't use it often enough to know off the top of my head. This line was moved from elsewhere in the code when I extracted a method to consolidate identical code that appeared 3-4 places in the class.

**Pull request merged**